File: Cap 12/Topico 12.6/Heranca Simples/herans.cs
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.6
 * 
 * Programa ..: HERANCA SIMPLES
*/

using System;

class CLS_SALA {
  public int SALA;
}

class CLS_ALUNO : CLS_SALA {
  public string NOME;
  public float[] NOTAS = new float[4];
  public float MEDIA;

  public float CMEDIA() {
    float SOMA = 0;
    for (int I = 0; I <= 3; I++) {
      SOMA = SOMA + NOTAS[I];
    }
    MEDIA = SOMA / 4;
    return MEDIA;
  }
}

public class Program {
	
  static CLS_ALUNO ALUNO = new CLS_ALUNO();
  
  static int I;

  public static void Main(string[] args) {
    
    Console.WriteLine("DADOS ESCOLARES");
    Console.WriteLine();
	
    Console.Write("Informe o nome: ");
    ALUNO.NOME = Console.ReadLine();
	
    Console.Write("Informe a sala: ");
    ALUNO.SALA = Convert.ToInt32(Console.ReadLine());
	
    Console.WriteLine("Informe as notas:");
    for (I = 0; I <= 3; I++) {
      Console.Write((I + 1) + "a. nota: ");
      ALUNO.NOTAS[I] = Convert.ToSingle(Console.ReadLine());
    }
	
    ALUNO.CMEDIA();

    Console.WriteLine();
    Console.WriteLine("RELATORIO ESCOLAR");
    Console.WriteLine();
	
    Console.WriteLine("Nome: " + ALUNO.NOME);
    Console.WriteLine("Sala: " + ALUNO.SALA);
    for (I = 0; I <= 3; I++) {
      Console.WriteLine($"{(I + 1)}a. nota: {ALUNO.NOTAS[I],5:F1}");
    }
    Console.WriteLine($"Media: {ALUNO.MEDIA,5:F1}");
	
  }
  
}

File: Cap 12/12.6 - Heranca/12.6.1 - Heranca Simples/herans.js
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.6
 * 
 * Programa ..: HERANCA SIMPLES
 * 
 */

class CLS_SALA {
  constructor() {
    this.SALA = 0;
  }
}

class CLS_ALUNO extends CLS_SALA {
  constructor() {
    super();
    this.NOME = "";
    this.NOTAS = [0, 0, 0, 0];
    this.MEDIA = 0;
  }

  CMEDIA() {
    let SOMA = 0;
    for (let I = 0; I <= 3; I++) {
      SOMA += this.NOTAS[I];
    }
    this.MEDIA = SOMA / 4;
    return this.MEDIA;
  }
}

const ALUNO = new CLS_ALUNO();

let I;
const readlineSync = require('readline-sync');

console.log("DADOS ESCOLARES\n");

ALUNO.NOME = readlineSync.question("Informe o nome: ");
ALUNO.SALA = parseInt(readlineSync.question("Informe a sala: "));
console.log("Informe as notas:");
for (I = 0; I <= 3; I++) {
  ALUNO.NOTAS[I] = parseFloat(readlineSync.question(`${I + 1}a. nota: `));
}

ALUNO.CMEDIA();

console.log("\nRELATORIO ESCOLAR\n");

console.log("Nome: " + ALUNO.NOME);
console.log("Sala: " + ALUNO.SALA);
for (I = 0; I <= 3; I++) {
  console.log(`${I + 1}a. nota: ` + ALUNO.NOTAS[I]);
}
console.log("Media: " + ALUNO.MEDIA.toFixed(1));
